uc.c

```diff
@@ -1036,6 +1036,16 @@ uc_err uc_hook_add(uc_engine *uc, uc_hook *hh, int type, void *callback,
         hook->insn = va_arg(valist, int);
         va_end(valist);
 
+
+#ifdef UNICORN_HAS_X86
```

Comment:
Would a function that calls out to validate the hook for a particular architecture be ok? 

For example something like:

#ifdef UNICORN_HAS_X86
if (arch == UC_ARCH_x86 && !x86_validate_instruction_hook(insn)){
 return error
}
#endif
#ifdef UNICORN_HAS_ARM
if(arch == UC_ARCH_ARM && !arm_validate_instruction_hook(insn){
...
#endif
....